.github/workflows/build-sphinx.yml

```diff
@@ -125,36 +127,46 @@ jobs:
             working-directory: 'dpnp/backend/doc'
 
       - name: Copy backend docs
-        run: cp -r dpnp/backend/doc/html doc/_build/html/backend_doc
+        run: cp -r dpnp/backend/doc/html ${{ env.PUBLISH_DIR }}/backend_doc
 
       # https://github.com/marketplace/actions/github-pages-action
+      # The step is only used to build docs while pushing PR to "master"
       - name: Deploy docs
         if: |
-          github.event.pull_request && !github.event.pull_request.head.repo.fork  &&
-          (github.ref == 'refs/heads/master' || (startsWith(github.ref, 'refs/heads/release') == true) || github.event_name == 'push' && contains(github.ref, 'refs/tags/'))
+          github.ref == 'refs/heads/master' && github.event_name == 'push' &&
+          github.event.ref == 'refs/heads/master' && github.event.repository && !github.event.repository.fork
         uses: peaceiris/actions-gh-pages@v3.9.3
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
           publish_dir: ${{ env.PUBLISH_DIR }}
+          keep_files: true
+          commit_message: ${{ github.event.head_commit.message }}
+          publish_branch: gh-pages
+          user_name: ${{ env.GITHUB_BOT_NAME }}
+          user_email: ${{ env.GITHUB_BOT_EMAIL }}
 
+      # The step is only used to build docs while pushing to PR branch
       - name: Publish pull-request docs
         if: |
-          github.event.pull_request && !github.event.pull_request.head.repo.fork && github.event.action != 'closed'
+          github.event_name == 'pull_request' && github.event.action != 'closed' &&
```

Comment:
`github.event.action != 'closed' && github.event.pull_request && !github.event.pull_request.head.repo.fork` is duplicated in three checks (151, 168, 181 lines). 
 It`d be better to create an env variable with this condition and use it in these checks

Reply:
I didn`t notice that the code being duplicated in different jobs.
Of course we can create an env variable in each jobs and avoid duplicating one check but I think it is not so necessary
Just an example of how I thought 
 
```
jobs:
  build-and-deploy:
    name: Build and Deploy Docs

    runs-on: ubuntu-20.04

    env:
      python-ver: '3.9'
      CHANNELS: '-c dppy/label/dev -c intel -c conda-forge --override-channels'
      ACTIVE_PR_NOT_FORK = ${{ github.event_name == 'pull_request' && github.event.action != 'closed' && github.event.pull_request && !github.event.pull_request.head.repo.fork }}
...
- name: Publish pull-request docs
        if: env.ACTIVE_PR_NOT_FORK
...
- name: Comment with URL to published pull-request docs
        if: env.ACTIVE_PR_NOT_FORK
...